include/mxnet/c_api.h

```diff
@@ -534,13 +554,34 @@ MXNET_DLL int MXNDArrayCreate(const mx_uint *shape,
  * \return 0 when success, -1 when failure happens
  */
 MXNET_DLL int MXNDArrayCreateEx(const mx_uint *shape,
-                              mx_uint ndim,
-                              int dev_type,
-                              int dev_id,
-                              int delay_alloc,
-                              int dtype,
-                              NDArrayHandle *out);
+                                mx_uint ndim,
+                                int dev_type,
+                                int dev_id,
+                                int delay_alloc,
+                                int dtype,
+                                NDArrayHandle *out);
 
+/*!
+ * \brief create a NDArray with specified shape and data type
+ * \param shape the pointer to the shape
+ * \param ndim the dimension of the shape
+ * \param dev_type device type, specify device we want to take
+ * \param dev_id the device id of the specific device
+ * \param delay_alloc whether to delay allocation until
+ *    the narray is first mutated
+ * \param dtype data type of created array
+ * \param name the name tag assigned
+ * \param out the returning handle
+ * \return 0 when success, -1 when failure happens
+ */
+MXNET_DLL int MXNDArrayCreateExWName(const mx_uint *shape,
+                                     mx_uint ndim,
+                                     int dev_type,
+                                     int dev_id,
+                                     int delay_alloc,
+                                     int dtype,
+                                     const char *name,
+                                     NDArrayHandle *out);
```

Comment:
I didnt see this use anywhere. Where is this called from ?

Reply:
It is called by the Python frontend API. In `ndarray/ndarray.py`, the original call to `MXNDArrayCreateEx` was replaced by `MXNDArrayCreateExWName`